firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -394,6 +396,25 @@ final class ToolbarMiddleware: FeatureFlaggable {
         store.dispatch(toolbarAction)
     }
 
+    private func updateReaderModeState(action: ToolbarMiddlewareAction,
+                                       state: AppState) {
+        guard let readerModeState = action.readerModeState,
+              let addressToolbarModel = generateAddressToolbarActions(action: action,
+                                                                      state: state,
+                                                                      lockIconImageName: action.lockIconImageName,
+                                                                      isEditing: false),
+              let navToolbarModel = generateNavigationToolbarActions(action: action,
```

Comment:
Can the navigation toolbar change when the reader mode state changes? If it can't then we shouldn't update it with the action.